src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DocumentPullDiagnosticsHandler.cs

```diff
@@ -68,12 +68,12 @@ public async Task<DiagnosticReport[]> HandleRequestAsync(DocumentDiagnosticsPara
 
             if (!_documentManager.TryGetDocument(request.TextDocument.Uri, out var documentSnapshot))
             {
-                return null;
+                return CreateNoDiagnosticsChangeResponse(request);
```

Comment:
If we fail to get the document, should we assume document is gone / report `null`?

Reply:
I'm going to assume yes, we report `null`/document gone in case document isn't found via the `DocumentManager`.